### PR TITLE
Logging for: create-initial-pr, create-initial-pr-comment, create-version-branch

### PR DESCRIPTION
**Pull request by Realtin**

### CODE REVIEW

File: jobs/create-initial-pr-comment.js [hunk 1]
```diff
@@ -44,7 +48,10 @@ module.exports = async function (
     number: prDoc.number
   }))
 
-  if (issue.state !== 'open' || issue.locked) return
+  if (issue.state !== 'open' || issue.locked) {
```
Comment (janl) [inline jobs/create-initial-pr-comment.js]: not sure if necessary, but maybe we want split the `||` out into two `if` statements, so we can log precisely why it doesn’t work.

File: jobs/create-version-branch.js [hunk 2]
```diff
@@ -34,26 +35,38 @@ module.exports = async function (
...
+  if (repository.private && (!billing || await getAccountNeedsMarketplaceUpgrade(accountId))) {
+    log.warn('exited: payment required')
+    return
+  }
 
   const [owner, repo] = repository.fullName.split('/')
   const config = getConfig(repository)
-  if (_.includes(config.ignore, dependency)) return
+  if (_.includes(config.ignore, dependency)) {
+    log.warn('exited: dependency ignored')
```
Comment (janl) [inline jobs/create-version-branch.js]: maybe `dependency ignored by user config`?

File: jobs/create-version-branch.js [hunk 3]
```diff
@@ -34,26 +35,38 @@ module.exports = async function (
...
+    return
+  }
   const installationId = installation.installation
   const ghqueue = githubQueue(installationId)
   const { default_branch: base } = await ghqueue.read(github => github.repos.get({ owner, repo }))
+  log.info('github: found default branch', {defaultBranch: base})
```
Comment (janl) [inline jobs/create-version-branch.js]: maybe `using default branch`? “found” implies that the finding was the problem, which it isn’t in this case.

File: jobs/create-version-branch.js [hunk 4]
```diff
@@ -64,9 +77,15 @@ module.exports = async function (
     }
 
     const oldPkgVersion = _.get(json, [type, dependency])
-    if (!oldPkgVersion) return
+    if (!oldPkgVersion) {
+      log.warn('exited: could not find old package version')
```
Comment (janl) [inline jobs/create-version-branch.js]: add the old package version to the log message

File: jobs/create-version-branch.js [hunk 5]
```diff
@@ -64,9 +77,15 @@ module.exports = async function (
...
+      return
+    }
 
-    if (semver.ltr(version, oldPkgVersion)) return // no downgrades
+    if (semver.ltr(version, oldPkgVersion)) { // no downgrades
+      log.warn('exited: would be a downgrade', {newVersion: version, oldVersion: oldPkgVersion})
```
Comment (janl) [inline jobs/create-version-branch.js]: like so! :)

File: jobs/create-version-branch.js [hunk 6]
```diff
@@ -133,7 +160,10 @@ module.exports = async function (
 
   // nothing to do anymore
   // the next action will be triggered by the status event
-  if (satisfies) return
+  if (satisfies) {
+    log.info('dependency satisfies version range')
```
Comment (janl) [inline jobs/create-version-branch.js]: maybe add `no action required`

File: jobs/create-version-branch.js [hunk 7]
```diff
@@ -181,6 +211,7 @@ module.exports = async function (
   })
 
   // verify pull requests commit
+  log.info('github: set greenkeeper/verify status')
```
Comment (janl) [inline jobs/create-version-branch.js]: maybe move this to after the `ghqueue.write`?

File: jobs/create-version-branch.js [hunk 8]
```diff
@@ -228,7 +266,7 @@ module.exports = async function (
   }
 }
 
-async function createPr ({ ghqueue, title, body, base, head, owner, repo }) {
+async function createPr ({ ghqueue, title, body, base, head, owner, repo, log }) {
```
Comment (janl) [inline jobs/create-version-branch.js]: why does this need `log` passed in?

Comment (Realtin) [reply jobs/create-version-branch.js]: function is outside of the module

Comment (janl) [reply jobs/create-version-branch.js]: doh :D

**Pull request merged**